### PR TITLE
Docs: Add workflows for tech docs

### PR DESCRIPTION
**Pull request by josmperez**
Add workflows for tech docs process -- see https://github.com/grafana/technical-documentation/wiki/Project-management#about-the-typedocs-label-workflow for details

### CODE REVIEW

File: .github/workflows/issue_commands.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+    {
+      "type": "label",
+      "name": "type/docs",
```
Comment (sympatheticmoose) [inline .github/workflows/issue_commands.json:4]: I'd guess this wont work since we don't have that label, we have `documentation`

**Review by sympatheticmoose — CHANGES_REQUESTED** [review]

File: .github/workflows/issue_commands.json (hunk 1 above)
Comment (jackw) [reply .github/workflows/issue_commands.json:4]: Yeah I don't think it'll work either. We need to add the label and probably best to then delete the `documentation` label.

Comment (jackw) [reply .github/workflows/issue_commands.json:4]: I believe it would be better if there is a convention in the docs team to use `type/docs` across repos.

Comment (sympatheticmoose) [reply .github/workflows/issue_commands.json:4]: done

File: .github/workflows/typedocs.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    types: [ labeled ]
+  pull_request:
+    types: [ labeled ]
+jobs:
+  main:
+    if: ${{ github.event.pull_request == false || github.event.pull_request.head.repo.full_name == github.repository }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Actions
+        uses: actions/checkout@v3
```
Comment (sympatheticmoose) [inline .github/workflows/typedocs.yml:13]: https://github.com/grafana/plugin-tools/pull/395 we should probably use 4 as we've just bumped this in another action

Comment (josmperez) [inline .github/workflows/typedocs.yml:13]:
```suggestion
        uses: actions/checkout@v4
```

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Review by sympatheticmoose — COMMENTED** [review]
We can do this, but it feels like overkill to me, its doing a lot just to add things to a project. Then its also 2 files being a mix of json/yaml.

I think the following would suffice in all honesty:

```
name: Add docs issues to project

on:
  pull_request:
    types:
      - labeled
  issues:
    types:
      - labeled
jobs:
  add-to-project:
    name: Add issue to project
    runs-on: ubuntu-latest
    steps:
      - uses: actions/add-to-project@v0.5.0
        with:
          project-url: https://github.com/orgs/grafana/projects/69
          github-token: ${{ secrets.ISSUE_COMMANDS_TOKEN }}
          labeled: type/docs
```
The token may have more privileges than it strictly requires, but if we're going to use it anyway 🤷

**Comment by josmperez** [comment]
> We can do this, but it feels like overkill to me

It's not clear to me how to take action on this comment. Can you please clarify?

**Review by sympatheticmoose — APPROVED** [review]
go ahead :)

**Pull request merged**